libraries/chain/transaction_context.cpp

```diff
@@ -131,6 +132,7 @@ namespace eosio { namespace chain {
          if( trx_specified_cpu_usage_limit <= objective_duration_limit ) {
             objective_duration_limit = trx_specified_cpu_usage_limit;
             billing_timer_exception_code = tx_cpu_usage_exceeded::code_value;
+            tx_cpu_usage_reason = tx_cpu_usage_exceeded_reason::user_specified_trx_max_cpu_usage_ms;
             _deadline = start + objective_duration_limit;
          }
       }
```

Comment:
all three instances of `_deadline = start + objective_duration_limit;` could be just moved after this `if`

Reply:
I think we should leave this as is to avoid additional diff for PR https://github.com/AntelopeIO/leap/pull/558.